quinn-proto/src/connection.rs

```diff
@@ -580,7 +580,27 @@ impl Connection {
         self.pending_acks.subtract(&info.acks);
     }
 
-    pub fn check_packet_loss(&mut self, now: u64) {
+    pub fn timeout(&mut self, now: u64, timer: Timer) -> bool {
+        match timer {
+            Timer::Close => {
+                self.io.timer_stop(Timer::Idle);
+                if !self.app_closed {
+                    self.state = State::Drained;
+                }
```

Comment:
This shouldn't be conditional; we are certainly in the `Drained` state now. The original code had a branch only because there wasn't much point updating the state one line above forgetting about it, but better to be explicit now that those lines are separated.

Reply:
Makes sense.